spec/models/travel_request_spec.rb

```diff
@@ -15,5 +16,131 @@
     it { is_expected.to respond_to :start_date }
     it { is_expected.to respond_to :status }
     it { is_expected.to respond_to :travel_category }
+    it { is_expected.to respond_to :pending? }
+    it { is_expected.to respond_to :canceled? }
+    it { is_expected.to respond_to :approved? }
+    it { is_expected.to respond_to :denied? }
+    it { is_expected.to respond_to :changes_requested? }
+  end
+
+  describe "#status" do
+    it "has a default of pending" do
+      expect(travel_request.status).to eq("pending")
+    end
+  end
+
+  describe "#approve!" do
+    let(:travel_request) { FactoryBot.create :travel_request, creator: user }
+    it "does not approve a pending event unless the approver is a department head" do
+      expect do
+        travel_request.approve!(agent: user)
+      end.to change(StateChange, :count).by(1)
+
+      expect(travel_request.status).to eq("pending")
+      expect(travel_request.persisted?).to be_truthy
+    end
+
+    it "approves a pending event when the approver is a department head" do
+      expect do
+        travel_request.approve!(agent: user.department.head)
+      end.to change(StateChange, :count).by(1)
+      expect(travel_request.status).to eq("approved")
+      expect(travel_request.persisted?).to be_truthy
+    end
+
+    it "Does not approve a change_request event" do
+      travel_request.change_request!(agent: user)
+      expect { travel_request.approve!(agent: user) }.to raise_error AASM::InvalidTransition
+    end
+
+    it "Does not approve a cancled event" do
+      travel_request.cancel!(agent: user)
+      expect { travel_request.approve!(agent: user) }.to raise_error AASM::InvalidTransition
+    end
+
+    it "Does not approve a denied event" do
+      travel_request.deny!(agent: user)
+      expect { travel_request.approve!(agent: user) }.to raise_error AASM::InvalidTransition
+    end
+  end
+
+  describe "#deny!" do
+    let(:travel_request) { FactoryBot.create :travel_request }
+    it "denies a pending event" do
+      expect do
+        travel_request.deny!(agent: user)
```

Comment:
Should only supervisors or dept heads be able to deny? I *think* this user is just at the staff level?